server/test/services/export/CsvExporterServiceTest.java

```diff
@@ -700,6 +703,47 @@ public void getDemographicsCsv_whenSubmitterIsApplicant_TiFieldsAreNotSet() thro
     assertThat(record.get("TI Organization")).isEmpty();
   }
 
+  // TODO(#9212): There should never be duplicate entries because question paths should be unique,
+  // but due to #9212 there sometimes are. They point at the same location in the applicant data so
+  // it doesn't matter which one we keep. Remove this test after this is fixed.
+  @Test
+  public void getProgramAllVersionsCsv_exportDoesNotFailWithDuplicateQuestion() throws Exception {
+    var questionOne =
+        testQuestionBank.maybeSave(
+            new TextQuestionDefinition(
+                QuestionDefinitionConfig.builder()
+                    .setName("dupe question 1")
+                    .setDescription("")
+                    .setQuestionText(LocalizedStrings.of())
+                    .setQuestionHelpText(LocalizedStrings.empty())
+                    .build()),
+            LifecycleStage.ACTIVE);
+    var questionTwo =
+        testQuestionBank.maybeSave(
+            new TextQuestionDefinition(
+                QuestionDefinitionConfig.builder()
+                    .setName("dupe question 2")
+                    .setDescription("")
+                    .setQuestionText(LocalizedStrings.of())
+                    .setQuestionHelpText(LocalizedStrings.empty())
+                    .build()),
+            LifecycleStage.ACTIVE);
+
+    ProgramModel fakeProgram =
+        FakeProgramBuilder.newActiveProgram()
+            .withQuestion(questionOne)
+            .withQuestion(questionTwo)
+            .build();
+    FakeApplicationFiller.newFillerFor(fakeProgram)
+        .answerTextQuestion(questionOne, "answer one")
+        .answerTextQuestion(questionTwo, "answer two")
+        .submit();
+
+    CSVRecord record = getParsedRecords(fakeProgram.id).get(0);
+
```

Comment:
**suggestion:** Can we ensure that it outputs the original question / answer too?

Reply:
It can't output the original question because it's been overwritten in the applicant data.